src/main/java/org/fogbowcloud/manager/core/datastore/DatabaseManager.java

```diff
@@ -0,0 +1,99 @@
+package org.fogbowcloud.manager.core.datastore;
+
+import org.apache.log4j.Logger;
+import org.fogbowcloud.manager.core.datastore.orderstorage.*;
+import org.fogbowcloud.manager.core.models.linkedlists.SynchronizedDoublyLinkedList;
+import org.fogbowcloud.manager.core.models.orders.Order;
+import org.fogbowcloud.manager.core.models.orders.OrderState;
+
+import java.sql.SQLException;
+
+public class DatabaseManager implements StableStorage {
+
+    private static final Logger LOGGER = Logger.getLogger(DatabaseManager.class);
+
+    private static DatabaseManager instance;
+
+    private ComputeOrderStorage computeOrderStorage;
+    private NetworkOrderStorage networkOrderStorage;
+    private VolumeOrderStorage volumeOrderStorage;
+    private AttachmentOrderStorage attachmentOrderStorage;
+    private OrderTimestampStorage orderTimestampStorage;
+
+    private DatabaseManager() throws SQLException {
+        this.computeOrderStorage = new ComputeOrderStorage();
+        this.networkOrderStorage = new NetworkOrderStorage();
+        this.volumeOrderStorage = new VolumeOrderStorage();
+        this.attachmentOrderStorage = new AttachmentOrderStorage();
+        this.orderTimestampStorage = new OrderTimestampStorage();
+    }
+
+    public static DatabaseManager getInstance() {
+        if (instance == null) {
+            try {
+                instance = new DatabaseManager();
+            } catch (SQLException e) {
+                LOGGER.error("Error instantiating database manager", e);
+                throw new Error(e);
```

Comment:
We now have a recipe about how to throw exception. We can throw a specified subset. Are you following this? In other words, are we allowed to throw this "new Error" exception?

Reply:
Using `FatalErrorException` now, since it is a runtime exception.